src/components/Photo/DocumentAutoCapture.js

```diff
@@ -0,0 +1,133 @@
+// @flow
+import * as React from 'react'
+import { h, Component } from 'preact'
+import Visibility from 'visibilityjs'
+import { screenshot } from '../utils/camera.js'
+import { DocumentOverlay } from '../Overlay'
+import Camera from '../Camera'
+import CameraError from '../CameraError'
+import { randomId } from '../utils/string'
+import { postToBackend } from '../utils/sdkBackend';
+
+const maxAttempts = 3
+
+const serverError = { name: 'SERVER_ERROR', type: 'error' }
+
+type Capture = {
+  id: string,
+  base64: string,
+  valid?: boolean,
+  processed?: boolean,
+}
+
+type State = {
+  hasError: boolean,
+  captures: Capture[],
+}
+
+type Props = {
+  token: string,
+  onValidCapture: Function,
+  onError: Function,
+  i18n: Object,
+  renderFallback: Function,
+  trackScreen: Function,
+  useFullScreen: Function,
+}
+
+export default class DocumentAutoCapture extends Component<Props, State> {
+  webcam = null
+
+  interval: ?Visibility
+
+  state: State = {
+    hasError: false,
+    captures: [],
+  }
+
+  componentDidMount () {
+    this.start()
+  }
+
+  componentWillUnmount () {
+    this.stop()
+  }
+
+  screenshot = () => screenshot(this.webcam, this.handleScreenshot)
+
+  start() {
+    this.stop()
+    this.interval = Visibility.every(1000, this.screenshot)
+  }
+
+  stop() {
+    Visibility.stop(this.interval)
+  }
+
+  handleScreenshot = (blob: Blob, base64: string) => {
+    if (this.unprocessed().length < maxAttempts) {
+      const id = randomId()
+      const capture: Capture = { id, base64 }
+      this.setState({
+        captures: [capture, ...this.state.captures].slice(0, maxAttempts),
+      })
+      this.validate(base64, id, valid =>
+        valid ? this.props.onValidCapture({ blob, base64, id }) : null
+      )
+    } else {
+      console.warn('Server response is slow, waiting for responses before uploading more')
+    }
+  }
+
+  unprocessed = (): Capture[] => this.state.captures.filter(({ processed }) => !processed)
+
+  failed = (): Capture[] => this.state.captures.filter(({ valid, processed }) => processed && !valid)
+
+  validate = (base64: string, id: string, callback: Function) => {
+    const { token } = this.props
+    const data = JSON.stringify({ image: base64, id })
+    postToBackend(data, token, ({ valid }) => {
+      this.setProcessed(id, !!valid)
+      callback(valid)
+    }, this.handleValidationError)
+  }
+
+  setProcessed(id: string, valid: boolean) {
+    const { captures } = this.state
+    const update = { valid: !!valid, processed: true }
+    this.setState({
+      captures: captures.map(capture => capture.id === id ? ({ ...capture, ...update }) : capture),
+    }, () => {
+      if (this.failed().length >= maxAttempts) {
```

Comment:
I don't think this reflects the original behaviour. I think we were just returning a console log after 3 attempts but no explicit error to the user. This will return an error to the user.

Reply:
Yes, @stefaniacardenas I think you are right.

I might be confused, but it looks to me like that the original intention is to display an error when all captures have been processed and none of them are valid.

In the current `capture.js` component (https://github.com/onfido/onfido-sdk-ui/blob/development/src/components/Capture/capture.js#L47)

```
componentWillReceiveProps(nextProps) {
  const {validCaptures, unprocessedCaptures, allInvalid} = nextProps
  ...
  if (allInvalid) this.onFileGeneralError()
}
``` 

And in `selectors.js` (https://github.com/onfido/onfido-sdk-ui/blob/development/src/core/store/selectors/index.js#L16):

```
export const allInvalidCaptureSelector = createSelector(
  currentCaptures,
  captures => captures.length > 0 && captures.every(c => c.processed && !c.valid)
)
```

Reply:
I see. Yes , that's confusing.  I think the concept of valid/invalid capture is legacy. I'm glad we got rid of it!

Reply:
@stefaniacardenas, should I just remove the functionality to display an error to the user when all captures are invalid then?

Reply:
Yes, please. I don't think we should display an error to the user.